src/test/java/de/cronn/validation_files_diff/action/ValidationFileAutomaticRenamerTest.java

```diff
@@ -0,0 +1,158 @@
+import com.intellij.psi.impl.JavaPsiFacadeEx;
+import com.intellij.refactoring.rename.RenameProcessor;
+import com.intellij.refactoring.rename.naming.AutomaticRenamerFactory;
+import com.intellij.testFramework.HeavyPlatformTestCase;
+import com.intellij.testFramework.PsiTestUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static de.cronn.assertions.validationfile.TestData.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ValidationFileAutomaticRenamerTest extends HeavyPlatformTestCase {
+
+	public void testRenameClass_renamesAllValidationFiles() throws IOException {
+		Path projectDir = createDefaultJavaModuleStructure();
+
+		Path openedTestFile = projectDir.resolve("src/test/Test.java");
+		Files.createFile(openedTestFile);
+		Files.writeString(openedTestFile, "public class Test {}");
+
+		Path validationFileDirectory = getValidationFileDirectory(projectDir);
+		Path outputFileDirectory = getOutputFileDirectory(projectDir);
+		Path tempFileDirectory = getTempFileDirectory(projectDir);
+
+		Files.createDirectories(validationFileDirectory);
+		Files.createDirectories(outputFileDirectory);
+		Files.createDirectories(tempFileDirectory);
+
+		createValidationFileInDirectories(projectDir, "Test_findStuff.json");
+
+		refreshFileSystem(projectDir);
+
+		PsiClass testClass = JavaPsiFacadeEx.getInstanceEx(getProject()).findClass("Test");
+
+		executeRename(testClass, "Test2");
+
+		assertValidationFileExistsInDirectories(projectDir, "Test2_findStuff.json");
+		assertValidationFileNotExistsDirectories(projectDir, "Test_findStuff.json");
+	}
+
+	public void testRenameMethod_renamesAllValidationFiles() throws IOException {
+		Path projectDir = createDefaultJavaModuleStructure();
+
+		Path openedTestFile = projectDir.resolve("src/test/Test.java");
+		Files.createFile(openedTestFile);
+		Files.writeString(openedTestFile, "public class Test {\n" +
+				"\tpublic void testSomething() {" +
+				"}\n" +
+				"}");
+
+		Path validationFileDirectory = getValidationFileDirectory(projectDir);
+		Path outputFileDirectory = getOutputFileDirectory(projectDir);
+		Path tempFileDirectory = getTempFileDirectory(projectDir);
+
+		Files.createDirectories(validationFileDirectory);
+		Files.createDirectories(outputFileDirectory);
+		Files.createDirectories(tempFileDirectory);
+
+		String validationFileName = "Test_testSomething.json";
+		String newValidationFileName = "Test_testSomething2.json";
+
+		createValidationFileInDirectories(projectDir, validationFileName);
+
+		refreshFileSystem(projectDir);
+
+		PsiMethod testMethod = JavaPsiFacadeEx.getInstanceEx(getProject()).findClass("Test").findMethodsByName("testSomething", false)[0];
+
+		executeRename(testMethod, "testSomething2");
+
+		assertValidationFileExistsInDirectories(projectDir, newValidationFileName);
+		assertValidationFileNotExistsDirectories(projectDir, validationFileName);
```

Comment:
Nice tests!

Do we also want to verify that existing content of the validation files stay the same and are unchanged?

Reply:
I fear, I had to extend the feature to have it useable for my project to ;)

Reply:
Fine for me, but the original questions about verifying that existing content remains unchanged is still open :D

Reply:
Since this dialog only suggests validation files to rename I don't think it is really necessary. 

But idc, done in d0b3c439fb127e37752d4b335d61798f9e957243